cli/tools/registry/pm/outdated/interactive.rs

```diff
@@ -0,0 +1,333 @@
+}
+
+#[derive(Debug)]
+struct State {
+  packages: Vec<PackageInfo>,
+  currently_selected: usize,
+  checked: HashSet<usize>,
+
+  name_width: usize,
+  current_width: usize,
+  start_row: u16,
+}
+
+impl State {
+  fn new(packages: Vec<PackageInfo>) -> anyhow::Result<Self> {
+    let name_width = packages
+      .iter()
+      .map(|p| p.name.len() + p.kind.scheme().len() + 1)
+      .max()
+      .unwrap_or_default();
+    let current_width = packages
+      .iter()
+      .map(|p| p.current_version.len())
+      .max()
+      .unwrap_or_default();
+
+    Ok(Self {
+      packages,
+      currently_selected: 0,
+      checked: HashSet::new(),
+
+      name_width,
+      current_width,
+      start_row: cursor::position()?.1,
+    })
+  }
+
+  fn render<W: std::io::Write>(&self, out: &mut W) -> anyhow::Result<()> {
+    use cursor::MoveTo;
+    use style::Print;
+    use style::PrintStyledContent;
+
+    crossterm::queue!(
+      out,
+      MoveTo(0, self.start_row),
+      terminal::Clear(terminal::ClearType::FromCursorDown),
+      PrintStyledContent("?".blue()),
+      Print(" Select which packages to update (<space> to select, ↑/↓/j/k to navigate, enter to accept, <Ctrl-c> to cancel)")
+    )?;
+
+    let base = self.start_row + 1;
+
+    for (i, package) in self.packages.iter().enumerate() {
+      if self.currently_selected == i {
+        crossterm::queue!(
+          out,
+          MoveTo(0, base + (self.currently_selected as u16)),
+          PrintStyledContent("❯".blue()),
+          Print(' '),
+        )?;
+      }
+      let checked = self.checked.contains(&i);
+      let selector = if checked { "●" } else { "○" };
+      crossterm::queue!(
+        out,
+        MoveTo(2, base + (i as u16)),
+        Print(selector),
+        Print(" "),
+      )?;
+
+      if self.currently_selected == i {
+        out.queue(style::SetStyle(
+          style::ContentStyle::new().on_black().white().bold(),
+        ))?;
+      }
+      let want = &package.new_version;
+      let new_version_highlight =
+        highlight_new_version(&package.current_version, want)?;
+      let formatted_name = format!(
+        "{}{}",
+        deno_terminal::colors::gray(format!("{}:", package.kind.scheme())),
+        package.name
+      );
+      let name_pad = " ".repeat(self.name_width + 2 - (package.name.len() + 4));
+
+      crossterm::queue!(
+        out,
+        Print(format!(
+          "{formatted_name}{name_pad} {:<current_width$} -> {}",
+          package.current_version,
+          new_version_highlight,
+          current_width = self.current_width
+        )),
+      )?;
+      if self.currently_selected == i {
+        out.queue(style::ResetColor)?;
+      }
+    }
+
+    out.queue(MoveTo(0, base + self.packages.len() as u16))?;
+
+    out.flush()?;
+
+    Ok(())
+  }
+}
+
+enum VersionDifference {
+  Major,
+  Minor,
+  Patch,
+}
+
+struct VersionParts {
+  major: u64,
+  minor: u64,
+  patch: u64,
+  pre: Option<String>,
```

Comment:
Should we be using `deno_semver::Version` here? https://docs.rs/deno_semver/0.7.1/deno_semver/struct.Version.html